### PR TITLE
Core tests: shape

### PR DESCRIPTION
**Pull request by XLegland**
Urgent:

- test creating multiple shapes (using 1 color) , and see if the global palette always contains 1 color (and not more)

### CODE REVIEW

**Review by M0UTOO — CHANGES_REQUESTED** [review]
There are some minors changes, but some of it are maybe not even required. Apart from that the rest of if lgtm 👍

File: core/tests/test_shape.h [hunk 1]
```diff
@@ -9,6 +9,370 @@
...
+    ok = shape_retain((Shape *const)s);
+    TEST_CHECK(ok);
+
+    shape_release((Shape *const)s);
+    shape_release((Shape *const)s);
+    shape_free((Shape *const)s);
+}
+
+// check that shape_release does not crash
+void test_shape_release(void) {
```
[the remark this answers is not in the document]
Comment (XLegland) [reply core/tests/test_shape.h:233]: Actually the shape is freed at the 2nd release, it releases its transform and then the transform frees its shape, in `_transform_free`:
```c
    switch (transform_get_type(t)) {
        case ShapeTransform: // Lua Shapes freed here
            shape_free((Shape *)transform_get_ptr(t));
            t->ptr = NULL;
            break;
        default:
            break;
    }
```

**Review by M0UTOO — APPROVED** [review]

**Pull request merged**